src/core/friendly_errors/sketch_verifier.js

```diff
@@ -111,16 +164,116 @@ function sketchVerifier(p5, fn) {
+      const message = `${errorType} "${name}" on line ${line} is being redeclared and conflicts with a p5.js ${errorType.toLowerCase()}. JavaScript does not declaring a ${errorType.toLowerCase()} more than once. p5.js reference: ${url}.`;
+      return message;
+    }
+
+    // Helper function that checks if a user definition has already been defined
+    // in the p5.js library, either as a constant or as a function.
+    function checkForRedefinition(name, libValue, line, type) {
+      try {
+        const userValue = eval("name");
+        if (libValue !== userValue) {
+          let message = generateFriendlyError(type, name, line);
+          console.log(message);
+          return true;
+        }
+      } catch (e) {
+        // If eval fails, the function hasn't been redefined
+        return false;
+      }
+      return false;
+    }
+
+    // Checks for constant redefinitions.
+    for (let { name, line } of allDefinitions) {
+      const libDefinition = constants[name];
+      if (libDefinition !== undefined) {
+        if (checkForRedefinition(name, libDefinition, line, "Constant")) {
+          return true;
+        }
+      }
+    }
+
+    // Get the names of all p5.js functions which are available globally
+    const classesWithGlobalFns = ['Renderer', 'Renderer2D', 'RendererGL'];
+    const globalFunctions = new Set(
+      classesWithGlobalFns.flatMap(className =>
+        Object.keys(p5Constructors[className]?.prototype || {})
+      )
+    );
+
+    for (let { name, line } of allDefinitions) {
+      if (!ignoreFunction.includes(name) && globalFunctions.has(name)) {
+        for (let className of classesWithGlobalFns) {
+          const prototypeFunc = p5Constructors[className]?.prototype[name];
+          if (prototypeFunc && checkForRedefinition(name, prototypeFunc, line, "Function")) {
+            return true;
+          }
+        }
+      }
+    }
+
+    // Additional check for other p5 constructors
+    const otherP5ConstructorKeys = Object.keys(p5Constructors).filter(
+      key => !classesWithGlobalFns.includes(key)
+    );
+    for (let { name, line } of allDefinitions) {
+      for (let key of otherP5ConstructorKeys) {
+        if (p5Constructors[key].prototype[name] !== undefined) {
+          const prototypeFunc = p5Constructors[key].prototype[name];
+          if (prototypeFunc && checkForRedefinition(name, prototypeFunc, line, "Function")) {
+            return true;
+          }
+        }
+      }
+    }
+
+    return false;
+  }
+
   fn.run = async function () {
     const userCode = await fn.getUserCode();
     const userDefinedVariablesAndFuncs = fn.extractUserDefinedVariablesAndFuncs(userCode);
 
-    return userDefinedVariablesAndFuncs;
+    if (fn.checkForConstsAndFuncs(userDefinedVariablesAndFuncs)) {
+      return;
+    }
   }
 }
 
 export default sketchVerifier;
 
 if (typeof p5 !== 'undefined') {
   sketchVerifier(p5, p5.prototype);
+  p5.prototype.loadP5Constructors();
```

Comment:
If we ever want to support loading this module separately, where `p5` may not be globally defined and the exported `sketchVerifier` function is run manually later, it looks like this `loadConstructors` call wouldn't get run. Maybe if we run this as the last thing inside of `sketchVerifier` then it'll get run either way?

Reply:
It is relatively unlikely this block will be run unless someone is importing this module as an addon library (which is what it's for). It should be moved into the `sketchVerifier` function instead or if deferred execution is desired, be set to run in the `presetup` lifecycle hook.

Reply:
Moved `loadP5Constructors` to a utils file!